test/float16.jl

```diff
@@ -14,6 +16,25 @@ g = Float16(1.)
 @test isless(g, f)
 @test !isless(f, g)
 
+@test convert(Bool,Float16(0.)) == false
+@test convert(Bool,Float16(0.1)) == true
```

Comment:
Maybe a holdover from when conversion to Bool was `x != 0`, which in hindsight was a bad idea.